var/spack/repos/builtin/packages/mapnik/package.py

```diff
@@ -15,11 +15,12 @@ class Mapnik(AutotoolsPackage):
     homepage = "https://mapnik.org/"
     url      = "https://github.com/mapnik/mapnik/releases/download/v3.0.22/mapnik-v3.0.22.tar.bz2"
 
+    version('3.0.23', sha256='4b1352e01f7ce25ab099e586d7ae98e0b74145a3bf94dd365cb0a2bdab3b9dc2')
     version('3.0.22', sha256='930612ad9e604b6a29b9cea1bc1de85cf7cf2b2b8211f57ec8b6b94463128ab9')
 
     depends_on('python', type=('build', 'run'))
-    # Build fails with boost@1.70
-    depends_on('boost@:1.69.0+regex+filesystem+system+icu+program_options cxxstd=11')
+    # Build fails with boost@1.73.0
+    depends_on('boost@:1.72.0 +regex+filesystem+system+icu+program_options cxxstd=11')
```

Comment:
Is this true also for 3.0.22? If not we can try to factor non-conditional requirements in their own statement:
```python
depends_on('boost +regex+filesystem+system+icu+program_options cxxstd=11')
depends_on('boost@:1.72.0', when='@3.0.23')
depends_on('boost@:1.69.0', when='@3.0.22')
```
Would this work for you?

Reply:
I got rid of the first line since it leads to the usual concretisation error.